build.gradle.kts

```diff
@@ -123,18 +127,24 @@ val gitVersion: String by lazy {
 }
 
 buildConfig {
-    appName = "mongo-kafka"
-    version = gitVersion
-    clsName = "Versions"
-    packageName = "com.mongodb.kafka.connect"
+    className("Versions")
+    packageName("com.mongodb.kafka.connect")
+    useJavaOutput()
+    buildConfigField("String", "NAME", "\"mongo-kafka\"")
+    buildConfigField("String", "VERSION", provider { "\"${gitVersion}\"" })
 }
 
 /*
  * Testing
  */
+tasks.register<Test>("testsOnJava8") {
+    javaLauncher.set(javaToolchains.launcherFor {
+        languageVersion.set(JavaLanguageVersion.of(8))
```

Comment:
Added a new test task (unit tests) running on Java 8

Reply:
How does it find Java 8?

Reply:
Its automatic - but we provide hints via the gradle.property and the JDK8 variables

Reply:
I'm just trying to wrap my head around this change.  I've always thought the build scripts should be agnostic to JDK version, so it looks weird to me that we have a task like this.  Is there a way to do it more flexibly, perhaps similar to what we do in the driver with jdkHome?  Something like:

```
./gradlew -PwithJDKVersion=8 test
```

and then set the language version based on that property (or leave it unset if the property is not specified).

Reply:
There is some coupling as we have to pick a version to compile against, I also provide hints as to where the java versions are installed in via the JDK8/JDK17 variables.

That said I like continuing to use the matrix and we already dynamically set the test jdkHome - so I've used that process but updated it to the toolchain syntax. :+1: